dvc/external_repo.py

```diff
@@ -69,6 +70,22 @@ def _external_repo(url=None, rev=None, cache_dir=None):
 
     repo = Repo(new_path)
     try:
+        # check if the URL is local and no default remote is present
+        # add default remote pointing to the original repo's cache location
+        if os.path.isdir(url):
+            rconfig = RemoteConfig(repo.config)
+            if not _default_remote_set(rconfig):
```

Comment:
@maykulkarni @efiop does this logic mean that if there's a default remote setup, still DVC skips the local cache? Because I think it should try the local cache first regardless of remotes.

Reply:
@jorgeorpinel Yes, precisely. It is hard for us to try both right now, as we don't have that kind of cascading remote logic implemented. I would wait for someone to ask for this.

Reply:
Sorry, I'm not sure I understand:

1. Every DVC repo has a cache directory right? So in theory it can always be the first place to get stuff from when using get/import from another local project. Why does this require complicated remote checks?

2. Does this PR really close #2599 per that issue's description and discussion?

Reply:
1. Yes, but "first place to get stuff from" means that there is a second place, and our code is not ready to cascade from first to second yet, if it can't find all the stuff it needs in the local cache dir.

2. From what we've understood initially - yes, but now it looks like we didn't quite understand each other.

Reply:
I see. OK!